docker/Dockerfile.local

```diff
@@ -1,34 +1,66 @@
-FROM nvidia/cuda:11.3.1-devel-ubuntu20.04
-ENV DEBIAN_FRONTEND=noninteractive
-ENV HOME=/root
+ARG DEVICE=gpu
+
+FROM nvidia/cuda:12.1.0-runtime-ubuntu20.04 as branch-gpu
+ENV dev_type=GPU
+
+RUN apt update && apt install -y --no-install-recommends \
+    python3-dev \
+    python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install DGL and Pytorch for GPU
+RUN pip3 install torch==2.1.2 \
+    && rm -rf /root/.cache
+
+RUN pip3 install networkx==3.1 pydantic dgl==2.1.0+cu121 \
+    -f https://data.dgl.ai/wheels/cu121/repo.html && rm -rf /root/.cache
+
+FROM public.ecr.aws/ubuntu/ubuntu:22.04_stable as branch-cpu
 
-RUN apt-get update
-RUN apt-get install -y --no-install-recommends build-essential python3-dev make g++ vim net-tools
+# Install DGL and Pytorch for CPU
+ENV dev_type=CPU
 
-RUN apt-get install -y python3-pip git wget psmisc
-RUN apt-get install -y cmake
+RUN apt update && apt install -y --no-install-recommends \
+    python3-dev \
+    python3-pip \
+    && rm -rf /var/lib/apt/lists/*
 
-# Install Pytorch
-RUN pip3 install networkx==3.1 pydantic
-RUN pip3 install torch==2.1.0+cu118 --extra-index-url https://download.pytorch.org/whl/cu118
+RUN pip3 install torch==2.1.2 --index-url https://download.pytorch.org/whl/cpu
+
+RUN pip3 install networkx==3.1 pydantic dgl==2.1.0 \
```

Comment:
I have met a problem for this, check issue here: https://github.com/awslabs/graphstorm/issues/809 It is the problem with DGL with version >= 2.0.0, but it has no problem with DGL == 1.1.3

Reply:
I have also created an issue on DGL repo: https://github.com/dmlc/dgl/issues/7333, we can use DGL 1.1.3 version first.

Reply:
Ok I'll try to reproduce this and can keep using DGL 1.1.x for now.

Reply:
I tried to repro this inside a CPU Docker container but haven't encountered this error. Do you know what yuo particular execution environment was?

Reply:
I think you can try that in any GPU docker container and you will meet that problem.

Reply:
Ah only happens in GPU envs, I got the error when running on a G5 with the GPU image.